### PR TITLE
Added deadzone to FollowCamera

### PR DESCRIPTION
**Pull request by JHelmsAtROBLOX**
This changes introduces a 45 degree zone around the camera's lookVector. If the player moves forward in that zone, the camera won't actually follow. Doesn't really effect things if you're using arrows to move, but with thumbstick makes possible to make more precise forward movements.

Relevant lines are 154-163

### CODE REVIEW

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx [hunk 1]
```diff
@@ -148,9 +149,18 @@ local function CreateFollowCamera()
 						end
 					elseif not (isInFirstPerson or userRecentlyPannedCamera) and not VRService.VREnabled then
 						local lastVec = -(self.LastCameraTransform.p - subjectPosition)
+						
 						local y = findAngleBetweenXZVectors(lastVec, self:GetCameraLook())
+						
+						-- This cutoff is to decide if humanoids angle of movement,
+						-- relative to the camera's look vector, is enough that
+						-- we wan the camera to be following them. The point is to provide
```
Comment (SolarCrane) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: looks like there are a couple typos in this description, perhaps it is in 'want'! :P

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx [hunk 2]
```diff
@@ -148,9 +149,18 @@ local function CreateFollowCamera()
...
 						local lastVec = -(self.LastCameraTransform.p - subjectPosition)
+						
 						local y = findAngleBetweenXZVectors(lastVec, self:GetCameraLook())
+						
+						-- This cutoff is to decide if humanoids angle of movement,
+						-- relative to the camera's look vector, is enough that
+						-- we wan the camera to be following them. The point is to provide 
+						-- a sizable deadzone to allow more precision forward movements.
+						local thetaCutoff = 0.4
+						local theta = y/timeDelta
```
Comment (SolarCrane) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: I am not sure if we check for the 0 timeDelta elsewhere. Sometimes we boot up the script and then run it without a step in between.

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx [hunk 3]
```diff
@@ -148,9 +149,18 @@ local function CreateFollowCamera()
...
+						
 						-- Check for NaNs
-						if IsFinite(y) and math.abs(y) > 0.0001 then
+						if IsFinite(y) and math.abs(y) > 0.0001 and math.abs(theta) > thetaCutoff then
```
Comment (SolarCrane) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: Have we checked how this works in VR? We may want to make sure it doesn't do weird behavior for that.

Comment (JHelmsAtROBLOX) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: Look at line 150, this only being called if not VRService.VREnabled

File: PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx (hunk 1 above)
Comment (JHelmsAtROBLOX) [inline PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera/FollowCamera.rbxmx]: Ha! Oops... I was in a rush this afternoon, I'll fix these, thanks.

**Pull request merged**